src/mmw/js/src/draw/views.js

```diff
@@ -64,7 +65,13 @@ function validateShape(polygon) {
     var area = coreUtils.changeOfAreaUnits(turfArea(polygon), 'm<sup>2</sup>', 'km<sup>2</sup>'),
         d = new $.Deferred();
 
-    if (area > MAX_AREA) {
+    var selfIntersectingShape = turfKinks(polygon).features.length > 0;
```

Comment:
I like that this is small and self contained, but consider looking into the `L.Draw::allowIntersection` options which may prevent a user from drawing a self intersected polygon at all - it may be more intuitive than accepting the shape, but then rejecting it.

https://github.com/Leaflet/Leaflet.draw#draw-handler-options